recipes/Text_to_Shell/Text_to_Shell.ipynb

```diff
@@ -42,24 +57,132 @@
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"My OS is {os_name()}\")"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "> **TIP:** If you are running on a Mac, it is likely that the invocation of `date` will be incorrect for the MacOS version of this command. Try adding the following sentence to the prompt and see what happens: `Make sure the generated shell commands are MacOS-compatible.`"
+    "Now let's write a helper function for running queries, wrapping the Ollama API. Note the default model name we'll used, Granite Code 20B (parameters). Also, note how we add additional content to the user's input prompt, such \"make sure you write code that works for _my_ system!\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def query(prompt: str, model: str = 'granite-code:20b') -> str:\n",
+    "    response = ollama.generate(\n",
+    "        model=model, \n",
+    "        prompt=f\"{prompt}. Make sure you generate code that is {os_name()}-compatible!\")\n",
+    "    print(response[\"response\"])\n",
+    "    return response[\"response\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Here's an example of a bash script that will achieve this task:\n",
+      "\n",
+      "```bash\n",
+      "#!/bin/bash\n",
+      "find . -type f -mtime -7 | head -n 50 | xargs ls -lt\n",
+      "```\n",
+      "\n",
+      "This script uses the `find` command to search for files (`-type f`) within the current working directory and its subdirectories. The `-mtime -7` option limits the search to files modified within the last week (i.e., 7 days). The `head -n 50` command selects the first 50 matching files, and `xargs ls -lt` displays the last modification time for each of these files.\n",
+      "\n",
+      "Note that this script assumes a Mac-based system as it uses the `ls` command with the `-lt` option to display the last modification time in a human-readable format. If you are using a different operating system, you may need to adjust the final command accordingly.\n"
+     ]
+    }
+   ],
+   "source": [
+    "query(\"\"\"\n",
+    "    Show me a bash script to print the first 50 files found under the current working directory\n",
+    "    that have been modified within the last week. Make sure you show the last modification time \n",
+    "    for each file in the output.\"\"\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Paste the command in the next cell. _**Keep the `!` shown**_, e.g., `!ls -al`, so Jupyter knows to run the command as a shell script instead of Python. (You can omit lines like `#!/bin/bash`.) \n",
+    "\n",
+    "Does the script work? If not try running the query again. Try modifying the query string, too."
```

Comment:
I like this idea of giving the user a cell to play with the generated code.

Reply:
You remind me that we have the "exec" notebook next; I'll add a comment that this is a "preview" of going to the next step, which is automatic execution.